packages/vitest/src/node/reporters/junit.ts

```diff
@@ -58,8 +58,9 @@ function escapeXML(value: any): string {
     true)
 }
 
-function getDuration(task: Task): string | undefined {
-  return task.result?.duration ? (task.result.duration / 1000).toFixed(10) : undefined
+export function getDuration(task: Task): string | undefined {
+  const duration = task.result?.duration ?? 0
+  return (duration / 1000).toFixed(10).replace(/^(\d+)((?:\.0+)|(\.\d+?)0+)$/, '$1$3')
```

Comment:
You can `parseFloat()` or cast to number to remove trailing zeroes. Using regex seems like an overkill for this

Reply:
Using `parseFloat` will result in exponential values.

Reply:
I guess we could use `Math.round(duration * 10e10) / 10e10` instead.

Reply:
Then lets keep it as a regex

Reply:
`String(Math.round(0.0000000000100 * 10e10) / 10e10)` this returns `"1e-11"`

Reply:
This might help? https://stackoverflow.com/a/50978675

Reply:
`0.0000000100.toLocaleString(undefined, {useGrouping:false, maximumFractionDigits:10 })` - `"0.00000001"` looks good